samtranslator/model/api/api_generator.py

```diff
@@ -555,6 +555,24 @@ def _openapi_postprocess(self, definition_body):
                 components['schemas'] = definition_body['definitions']
                 definition_body['components'] = components
                 del definition_body['definitions']
+
+            if definition_body.get("paths"):
+                for path in definition_body.get("paths"):
+                    if definition_body.get("paths").get(path).get("options"):
+                        definition_body_options = definition_body.get("paths").get(path).get("options").copy()
+                        for field in definition_body_options.keys():
+                            # remove unsupported produces and consumes in options for openapi3
+                            if field in ["produces", "consumes"]:
+                                del definition_body["paths"][path]["options"][field]
+                            # add schema for the headers in options section for openapi3
+                            if field in ["responses"]:
+                                headers = definition_body["paths"][path]["options"][field]['200']['headers']
```

Comment:
Are `200` and `headers` guaranteed to exist?

Reply:
Yes they are guranteed to exist if SAM generates the definition body/inline swagger. This is the part of the swagger definition sam generates: https://github.com/awslabs/serverless-application-model/blob/master/samtranslator/swagger/swagger.py#L361-L366

Reply:
Can you verify that this method doesn't run if SAM didn't generate the swagger?

Reply:
I will add a check for `200` and `headers` to make sure we won't throw `5xx` errors if the openapi template is incorrect